Source/WebCore/html/OffscreenCanvas.cpp

```diff
@@ -356,19 +356,11 @@ ExceptionOr<RefPtr<ImageBitmap>> OffscreenCanvas::transferToImageBitmap()
         auto buffer = allocateImageBuffer();
         if (!buffer)
             return { RefPtr<ImageBitmap> { nullptr } };
-
+        if (webGLContext->compositingResultsNeedUpdating())
+            webGLContext->prepareForDisplay();
         RefPtr gc3d = webGLContext->graphicsContextGL();
-        gc3d->drawSurfaceBufferToImageBuffer(GraphicsContextGL::SurfaceBuffer::DrawingBuffer, *buffer);
-
-        // FIXME: The transfer algorithm requires that the canvas effectively
-        // creates a new backing store. Since we're not doing that yet, we
-        // need to erase what's there.
-
-        GCGLfloat clearColor[4] { };
-        gc3d->getFloatv(GraphicsContextGL::COLOR_CLEAR_VALUE, clearColor);
-        gc3d->clearColor(0, 0, 0, 0);
-        gc3d->clear(GraphicsContextGL::COLOR_BUFFER_BIT | GraphicsContextGL::DEPTH_BUFFER_BIT | GraphicsContextGL::STENCIL_BUFFER_BIT);
-        gc3d->clearColor(clearColor[0], clearColor[1], clearColor[2], clearColor[3]);
+        gc3d->drawSurfaceBufferToImageBuffer(GraphicsContextGL::SurfaceBuffer::DisplayBuffer, *buffer);
+        webGLContext->markDrawingBuffersDirtyAfterTransfer();
```

Comment:
It might be nice to move most of these lines into a function on `WebGLRenderingContextBase`, so that `OffscreenCanvas` doesn't need to worry about the lower level WebGL details.

I guess that equally applies to the earlier code for 2D/ImageBitmap as well.

Reply:
Yes, polymorphic transferToNativeImage will be implemented once the dust settles for other polymorphic parts.